src/telemetry/telemetryService.ts

```diff
@@ -162,13 +157,9 @@ export default class TelemetryService {
         flushInterval: 10000 // 10 seconds is the default libraries' value.
       });
 
-      const identity: AnalyticsNodeIdentity = {
-        anonymousId: this._segmentAnonymousId
-      };
-      if (this._segmentUserId) {
-        identity.userId = this._segmentUserId;
-      }
-      this._segmentAnalytics.identify(identity);
+      const segmentProperties = this.getTelemetryUserIdentity();
+      this._segmentAnalytics.identify(segmentProperties);
+      log.info('TELEMETRY identify', segmentProperties);
```

Comment:
These changes seem unrelated, is it drive by cleanup?

Reply:
Yeah clean up to make it closed to the recent fix in mongosh: https://github.com/mongodb-js/mongosh/pull/1229/files#diff-ac2009df67ba6eafe8b274126601bb9491e31f6920c72bbd1b1c33d34864e237R78-R86